test/components/rankers/test_sentence_transformers_similarity.py

```diff
@@ -350,6 +351,25 @@ def test_score_threshold(self):
         out = ranker.run(query="test", documents=documents)
         assert len(out["documents"]) == 1
 
+    def test_scores_cast_to_python_float_when_numpy_scalars_returned(self):
+        mock_cross_encoder = MagicMock()
+        ranker = SentenceTransformersSimilarityRanker(model="model")
+        ranker._cross_encoder = mock_cross_encoder
+
+        # Simulate backend returning numpy scalar types
+        mock_cross_encoder.rank.return_value = [
+            {"score": np.float32(0.123), "corpus_id": 0},
+            {"score": np.float64(0.456), "corpus_id": 1},
+        ]
+
+        documents = [Document(content="doc 0"), Document(content="doc 1")]
+        out = ranker.run(query="test", documents=documents)
+
+        assert len(out["documents"]) == 2
+        for d in out["documents"]:
+            assert isinstance(d.score, float)
+            assert not isinstance(d.score, np.floating)
+
     @pytest.mark.integration
```

Comment:
thanks for the unit test!

Please also modify the following integration tests (`test_run`, `test_run_top_k`, and `test_run_single_document`) to also check that `doc.score` is a `float`